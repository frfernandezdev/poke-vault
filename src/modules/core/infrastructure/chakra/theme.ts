import { extendTheme, type ThemeConfig } from "@chakra-ui/react";

const config: ThemeConfig = {
	initialColorMode: 'system',
	useSystemColorMode: true
}

const theme = extendTheme({
	config,
	fonts: {
		heading: 'var(--font-poppins)',
		body: 'var(--font-poppins)'
	},
	colors: {
		brown: {
			400: '#d04848',
			600: '#8f2323'
		}
	}
});	

export default theme;
