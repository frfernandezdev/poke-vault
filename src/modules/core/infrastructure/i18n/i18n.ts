import { createInstance } from "i18next";
import LanguageDetector from 'i18next-browser-languagedetector';

const i18n = createInstance({
		fallbackLng: "en",
		debug: true,
		interpolation: {
			escapeValue: false 
		},
		resources: {
			en: {
				translation: import('../../../../../public/locales/en/common.json')		
			},
			es: {
				Translation: import('../../../../../public/locales/es/common.json')
			}
		},
	});

i18n.use(LanguageDetector);
i18n.init();

export default i18n;
