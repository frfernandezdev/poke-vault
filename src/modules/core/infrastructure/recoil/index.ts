import RecoilProvider from './RecoilProvider';
import { CoreAtom } from './atoms/CoreAtom';
import CoreDrawerSelector from './selectors/CoreDrawerSelector';
import CoreLoaderSelector from './selectors/CoreLoaderSelector';
import CoreSnackbarSelector from './selectors/CoreSnackbarSelector';
import CoreThemeSelector from './selectors/CoreThemeSelector';



export { 
	RecoilProvider,
	CoreAtom,
	CoreDrawerSelector,
	CoreLoaderSelector,
	CoreSnackbarSelector,
	CoreThemeSelector
}
