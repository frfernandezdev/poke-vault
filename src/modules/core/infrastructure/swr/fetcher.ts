import { AxiosRequestConfig } from "axios";
import instance from "../axios";

const fetcher = <D extends { [key: string]: any }>(
  config: AxiosRequestConfig<D>
) => instance.request(config).then(res => res.data);

export default fetcher;

