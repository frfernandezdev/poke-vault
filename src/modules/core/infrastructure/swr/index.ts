import SWRProvider from "./Provider";

export { SWRProvider };
