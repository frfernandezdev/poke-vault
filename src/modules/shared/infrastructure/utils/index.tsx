import alternativeColors from "./alternativeColors";

const bgColor = (color: string) => ["white", "black"].includes(color)
    ? alternativeColors[color]
    : `${color ?? 'gray'}.400`;

const bgColorHover = (color: string) => ["white", "black"].includes(color)
	? alternativeColors[color]
	: `${color ?? 'gray'}.600`;
const textColor = (color:string) => color === "white" ? "black" : "white";


export {
	alternativeColors,
	bgColor,
	bgColorHover,
	textColor
}

