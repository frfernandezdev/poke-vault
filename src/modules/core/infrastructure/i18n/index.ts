import i18n from './i18n';
import I18NProvider from './Provider';

export { i18n, I18NProvider };
