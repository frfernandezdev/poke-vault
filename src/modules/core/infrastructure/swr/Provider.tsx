"use client";

import { ReactNode } from "react";
import { SWRConfig } from "swr";
import fetcher from "./fetcher";

export default function SWRProvider({ children }: { children: ReactNode }) {
  return <SWRConfig value={{ fetcher }}>{children}</SWRConfig>;
}
