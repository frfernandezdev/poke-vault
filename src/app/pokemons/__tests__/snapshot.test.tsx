import { render } from '@testing-library/react'
import Page from "../page";
import { RecoilProvider } from "@/modules/core/infrastructure/recoil";
import { SWRProvider } from "@/modules/core/infrastructure/swr";
import { I18NProvider } from "@/modules/core/infrastructure/i18n";
import { ThemeProvider } from "@/modules/core/infrastructure/chakra";
 
it('renders homepage unchanged', () => {
  const { container } = render(
		<RecoilProvider>
			<SWRProvider>
				<ThemeProvider>
					<I18NProvider>
						<Page />
					</I18NProvider>
				</ThemeProvider>
			</SWRProvider>
		</RecoilProvider>
	);
  expect(container).toMatchSnapshot()
})
