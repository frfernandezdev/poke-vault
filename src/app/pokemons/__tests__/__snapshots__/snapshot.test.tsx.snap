// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <script
    id="chakra-script"
  >
    !(function(){try{var a=function(c){var v="(prefers-color-scheme: dark)",h=window.matchMedia(v).matches?"dark":"light",r=c==="system"?h:c,o=document.documentElement,s=document.body,l="chakra-ui-light",d="chakra-ui-dark",i=r==="dark";return s.classList.add(i?d:l),s.classList.remove(i?l:d),o.style.colorScheme=r,o.dataset.theme=r,r},n=a,m="system",e="chakra-ui-color-mode",t=localStorage.getItem(e);t?a(t):localStorage.setItem(e,a(m))}catch(a){}})();
  </script>
  <div
    class="css-0"
  >
    <div
      class="css-qdv6cs"
      style="height: 110px; box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0), 0 2px 4px -1px rgba(0, 0, 0, 0);"
    >
      <div
        class="chakra-container css-lv4due"
      >
        <div>
          <div
            class="css-70qvj9"
          >
            <button
              aria-label="menu"
              class="chakra-button css-vm9dlr"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <div
              class="css-1rr4qq7"
            >
              <h6
                class="chakra-heading css-v5bq0a"
                style="transform: translateY(46px) scale(1.3) translateZ(0);"
              >
                PokeVault
              </h6>
            </div>
            <button
              aria-label="menu"
              class="chakra-button css-vm9dlr"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-container css-vhc7ah"
      style="padding-top: 110px;"
    >
      <div
        class="infinite-scroll-component__outerdiv"
      >
        <div
          class="infinite-scroll-component "
          style="height: auto; overflow: auto;"
        >
          <div
            class="css-1hisz16"
          />
          <div
            class="css-1hisz16"
          >
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
            <a
              class="chakra-link chakra-card css-19lsh19"
              href="/pokemons/undefined"
            >
              <div
                class="chakra-card__body css-1idwstw"
              >
                <div
                  class="chakra-skeleton__group css-1d595ex"
                >
                  <div
                    class="chakra-skeleton css-1ysakuj"
                  />
                </div>
                <div
                  class="css-10iu0sy"
                >
                  <div
                    class="chakra-skeleton css-ykur7l"
                  />
                </div>
                <div
                  class="css-fyhvt3"
                >
                  <div
                    class="chakra-stack css-m3jj3s"
                  >
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                    <div
                      class="chakra-skeleton__group css-cvrs1r"
                    >
                      <div
                        class="chakra-skeleton css-1ysakuj"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
